extension/js/common/message-renderer.ts

```diff
@@ -431,7 +431,21 @@ export class MessageRenderer {
 
   private getMessageInfo = async (fullMsg: GmailRes.GmailMsg): Promise<MessageInfo> => {
     const sentDate = GmailParser.findHeader(fullMsg, 'date');
-    const sentDateStr = sentDate ? Str.fromDate(new Date(sentDate)).replace(' ', ' at ') : '';
+    const gmailDateSelector = 'div.gK span[title]';
+    // ensure that the attribute resembles a date by checking its time
+    const gmailDateReceived =
+      $(gmailDateSelector).length > 0 && !isNaN(new Date(String($(gmailDateSelector).attr('title'))).getTime()) ? $(gmailDateSelector).attr('title') : '';
+    // detects current timezone by checking if Gmail's dateTime includes trailing AM/PM otherwise its 24hour format
+    const ishour12Format = (gmailDateReceived?.endsWith('M') ? true : false) || false;
+    const formatOptions: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit',
+      hour: '2-digit',
+      minute: '2-digit',
+      hour12: ishour12Format,
+    };
+    const sentDateStr = sentDate ? new Date(sentDate).toLocaleString([], formatOptions).replace(', ', ' at ') : '';
```

Comment:
current implementation looks quite complicated, I think it'll be better to use your original version (where you parsed message date from `div.gK span[title]`) + add fallback using `.toLocaleString()`, so it'll be like this:
```ts
let messageDate = $('div.gK span[title]').attr('title');
if (!messageDate || isNaN(Date.parse(messageDate))) {
  const sentDate = GmailParser.findHeader(fullMsg, 'date');
  messageDate = sentDate ? new Date(sentDate).toLocaleString() : '';
}

Reply:
Thanks @sosnovsky - could you please check if the changes now is good to you? The only way to test the toLocaleString() is by checking its epoch date version.